src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DefaultProjectInstanceEvaluator.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.AspNetCore.Razor.OmniSharpPlugin
+{
+    [Shared]
+    [Export(typeof(ProjectInstanceEvaluator))]
+    public class DefaultProjectInstanceEvaluator : ProjectInstanceEvaluator
+    {
+        internal const string TargetFrameworkPropertyName = "TargetFramework";
+        internal const string TargetFrameworksPropertyName = "TargetFrameworks";
+        private const string CompileTargetName = "Compile";
+        private const string CoreCompileTargetName = "CoreCompile";
+        private const string RazorGenerateTargetName = "RazorGenerate";
+        private const string RazorComponentGenerateTargetName = "RazorComponentGenerate";
+        private static readonly IEnumerable<ILogger> EmptyMSBuildLoggers = Enumerable.Empty<ILogger>();
+        private readonly OmniSharpForegroundDispatcher _foregroundDispatcher;
+        private readonly object _evaluationLock = new object();
+        private ProjectCollection _projectCollection;
+
+        [ImportingConstructor]
+        public DefaultProjectInstanceEvaluator(OmniSharpForegroundDispatcher foregroundDispatcher)
+        {
+            if (foregroundDispatcher == null)
+            {
+                throw new ArgumentNullException(nameof(foregroundDispatcher));
+            }
+
+            _foregroundDispatcher = foregroundDispatcher;
+        }
+
+        public override ProjectInstance Evaluate(ProjectInstance projectInstance)
+        {
+            if (projectInstance == null)
+            {
+                throw new ArgumentNullException(nameof(projectInstance));
+            }
+
+            lock (_evaluationLock)
+            {
+                _foregroundDispatcher.AssertBackgroundThread();
+
+                var refreshTargets = new List<string>()
+                {
+                    // These are the default targets for the project instance that OmniSharp runs.
+                    CompileTargetName,
+                    CoreCompileTargetName
+                };
+
+                if (projectInstance.Targets.ContainsKey(RazorGenerateTargetName))
+                {
+                    refreshTargets.Add(RazorGenerateTargetName);
```

Comment:
Do we not care about the [`RazorTagHelper`](https://github.com/aspnet/AspNetCore-Tooling/blob/057d4f35705187a0b553f83c6c3e5e101acef5d1/src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.CodeGeneration.targets#L99) target here?

Reply:
Doesn't that get called as part of RazorGenerate?

Trying to pick top level targets so we can change things in Razor without functionally breaking VSCode editing

Reply:
> Doesn't that get called as part of RazorGenerate?

Yes. But I remember `RazorTagHelper` gets called in some instances without having to invoke `RazorGenerate`. But I'm not up to date in that area. Bringing in the expert @pranavkm

Reply:
Can you try this - https://github.com/aspnet/AspNetCore-Tooling/blob/master/src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.DesignTime.targets#L90?

Reply:
Those definitely work but it seems the component target is even available for 1.0 projects. Is that expected?

Reply:
Talked offline, this is expected.